Public/Src/Cache/DistributedCache.Host/Service/Internal/DualServerClientContentStore.cs

```diff
@@ -0,0 +1,172 @@
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="fileSystem">File system object used by the client.</param>
+        /// <param name="logger">Logger.</param>
+        /// <param name="copier">File copier used by the content server.</param>
+        /// <param name="pathTransformer">File path transformer used by the content server.</param>
+        /// <param name="maxQuotaMB">Maximum size (in MB) of the cache.</param>
+        /// <param name="cacheName">Name of the cache.</param>
+        /// <param name="grpcPort">GRPC port which the server listens to and the client transmits to.</param>
+        /// <param name="stampId">Redis stamp identifier.</param>
+        /// <param name="ringId">Redis ring identifier.</param>
+        /// <param name="scenarioName">Unique name for the CAS. Allows multiple CAS services to coexist.</param>
+        /// <param name="dataRootPath">Path to the root of CAS content.</param>
+        public DualServerClientContentStore(
+            IAbsFileSystem fileSystem,
+            ILogger logger,
+            IAbsolutePathFileCopier copier,
+            IAbsolutePathTransformer pathTransformer,
+            int maxQuotaMB,
+            string cacheName,
+            int grpcPort,
+            string stampId,
+            string ringId,
+            string scenarioName,
+            AbsolutePath dataRootPath)
+        {
+            Contract.Assert(grpcPort >= 0);
+            Contract.Assert(maxQuotaMB >= 0);
+            Contract.Assert(!string.IsNullOrWhiteSpace(cacheName));
+            Contract.Assert(!string.IsNullOrWhiteSpace(ringId));
+            Contract.Assert(!string.IsNullOrWhiteSpace(stampId));
+
+            var redisConnectionString = Environment.GetEnvironmentVariable(EnvironmentConnectionStringProvider.RedisConnectionStringEnvironmentVariable);
+
+            var localCasSettings = LocalCasSettings.Default(maxQuotaMB, dataRootPath.Path, cacheName, (uint)grpcPort);
+            localCasSettings.ServiceSettings.ScenarioName = scenarioName;
+
+            if (dataRootPath.IsLocal)
+            {
+                localCasSettings.DrivePreferenceOrder.Add(AbsolutePath.RootPath(dataRootPath.DriveLetter).Path);
+            }
+
+            var contentServerFactory = new ContentServerFactory(new DistributedCacheServiceArguments(
+                logger,
+                copier,
+                pathTransformer,
+                new TestDistributedCacheServiceHost(),
+                new HostInfo(stampId, ringId, new List<string>()),
+                default(CancellationToken),
+                dataRootPath.Path,
+                new DistributedCacheServiceConfiguration(
+                    localCasSettings,
+                    DistributedContentSettings.CreateEnabled(new Dictionary<string, string>() { { stampId, redisConnectionString } }, true)),

```

Comment:
>DistributedContentSettings.CreateEnabled(new Dictionary<string, string>() { { stampId, redisConnectionString } }, true)) [](start = 20, length = 120)

I was thinking a path to a settings file should be specified as a part of the config data (or embedded therein). Are you doing this to simply things for initial testing, or is this supposed to be the final state?